pkg/monitor/monitor/monitor_manager.go

```diff
@@ -427,3 +422,253 @@ func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef,
+		if err != nil {
+			klog.Errorf("Smooth migration for tm[%s/%s], fail to patch PV %s, err: %v", monitor.Namespace, monitor.Name, monitor.Status.DeploymentStorageStatus.PvName, err)
+			return false, err
+		}
+		// smooth migration successfully and clean status
+		monitor.Status.DeploymentStorageStatus = nil
+		return true, nil
+	}
+
+	if len(deploymentPvc.Spec.VolumeName) <= 0 {
+		klog.Errorf("Smooth migration for tm[%s/%s], old pvc not bind pv", monitor.Namespace, monitor.Name)
+		return false, nil
+	}
+
+	deploymentPv, err := m.deps.PVLister.Get(deploymentPvc.Spec.VolumeName)
+	if err != nil {
+		klog.Errorf("Smooth migration for tm[%s/%s], fail to get PV %s, err: %v", monitor.Namespace, monitor.Name, deploymentPvc.Spec.VolumeName, err)
+		return false, err
+	}
+	if deploymentPv.Spec.PersistentVolumeReclaimPolicy == corev1.PersistentVolumeReclaimDelete {
+		klog.Errorf("Smooth migration for tm[%s/%s], pv[%s] policy is delete, it must be retain", monitor.Namespace, monitor.Name, deploymentPvc.Spec.VolumeName)
+		return false, err
+	}
+
+	if deploymentPv.Spec.ClaimRef != nil && deploymentPv.Spec.ClaimRef.Name == firstStsPvcName {
+		// smooth migration successfully and clean status
+		monitor.Status.DeploymentStorageStatus = nil
+		return true, nil
+	}
+
+	// firstly patch status
+	if monitor.Status.DeploymentStorageStatus == nil {
+		err = m.patchTidbMonitorStatus(monitor, deploymentPvc.Spec.VolumeName)
+		if err != nil {
+			return false, err
+		}
+		//monitor patch status successfully
+	}
+
+	err = m.deps.PVCControl.DeletePVC(monitor, &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentPvcName,
+			Namespace: monitor.Namespace,
+		},
+	})
+	if err != nil {
+		klog.Errorf("Fail to delete the PVC %s for tm [%s/%s], err: %v", deploymentPvcName, monitor.Namespace, monitor.Name, err)
+		return false, err
+	}
+	err = m.patchPVClaimRef(deploymentPv, firstStsPvcName, monitor)
+	if err != nil {
+		klog.Errorf("Smooth migration for tm[%s/%s], fail to patch PV %s, err: %v", monitor.Namespace, monitor.Name, deploymentPvc.Spec.VolumeName, err)
+		return false, err
+	}
+	// smooth migration successfully and clean status
+	monitor.Status.DeploymentStorageStatus = nil
+	return true, nil
+}
+
+func (c *MonitorManager) validate(tidbmonitor *v1alpha1.TidbMonitor) bool {
+	errs := v1alpha1validation.ValidateTidbMonitor(tidbmonitor)
+	if len(errs) > 0 {
+		aggregatedErr := errs.ToAggregate()
+		klog.Errorf("tidbmonitor %s/%s is not valid and must be fixed first, aggregated error: %v", tidbmonitor.GetNamespace(), tidbmonitor.GetName(), aggregatedErr)
+		c.deps.Recorder.Event(tidbmonitor, corev1.EventTypeWarning, "FailedValidation", aggregatedErr.Error())
+		return false
+	}
+	return true
+}
+
+func (m *MonitorManager) syncTidbMonitorPV(tm *v1alpha1.TidbMonitor) error {
+	ns := tm.GetNamespace()
+	instanceName := tm.Name
+
+	l, err := label.New().Instance(instanceName).TiDBMonitor().Selector()
```

Comment:
Should use `buildTidbMonitorLabel` here?

Reply:
How to build labelSelector with buildTidbMonitorLabel return value?